Remotion/Data/DomainObjects/ReflectionUtility.cs

```diff
@@ -37,6 +39,18 @@ public static class ReflectionUtility
     private static readonly Type s_binaryPropertyValueType = typeof (byte[]);
     private static readonly Type s_typePropertyValueType = typeof (Type);
     private static readonly Type s_objectIDPropertyValueType = typeof (ObjectID);
+    private static readonly ConcurrentDictionary<Type, (bool CanAscribeTo, Type ItemType)> s_canAscribeToGenericArgumentsOfObjectListCache = new();
+
+    private static readonly Func<Type, ValueTuple<bool, Type>> s_cacheValueFactory =
+        static t =>
+        {
+          var canAscribeTo = t.CanAscribeTo (typeof (IReadOnlyList<>)) && t.CanAscribeTo (typeof (ObjectList<>));
```

Comment:
```suggestion
          var canAscribeTo = typeof (IReadOnlyList<DomainObject>).IsAssignableFrom (t) && t.CanAscribeTo (typeof (ObjectList<>));
```

The idea was that we can short curcuit the expesive check for types that arent actually in the running. And while I'm at it, we should use `typeof (DomainObjectCollection)` instead of `typeof (IReadOnlyList<DomainObject>)` here, since this the better/closer match. I probably didn't think of this the first time.